.github/workflows/no-duplicate-filenames.yml

```diff
@@ -0,0 +1,42 @@
+name: No duplicate filenames
+
+on:
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
+jobs:
+  no-duplicate-filenames:
+    name: No duplicate filenames
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Check if there are files with a duplicate name
+      run: |
+        set -e -o pipefail
```

Comment:
Please use `shell: bash` instead of this. GitHub promises to set all these things correctly when you explicitly request bash.